### ISSUE TITLE
Feature Request: Precache Media images.

### ISSUE THREAD

**Issue by grafixoner** [nextcloud/ios#658]
It would be great to have an option to pre-cache images for the media view (throttled) in the background.

I loaded about 20k images into Nextcloud, then spent 3 days generating previews for them.  I notice the media view downloads the thumbnails as needed and put a ton of load (DDOS) style to my server if I scrolled through them all.  

In order to work around this I scrolled the media view at my desk a page at a time for a couple days.  Once all the images are cached it works great.

**Comment by PnjDbq** [comment]
There's an app that might do what you are interested in: https://github.com/rullzer/previewgenerator

**Comment by grafixoner** [comment]
Yes, that app doesn't create the thumbs needed or something, because even after running that the Media view inside the iOS app tried creating thousands of thumbnails.

**Comment by Bellusterra** [comment]
I would like to contribute to this. I'm finding the same thing to be true as @grafixoner mentioned. the preview generator app works perfectly well on the server side generating the thumbnails. I believe the confusion is that the ios app still needs to "pull" down these thumbnails from the server when opening a folder or when the photos are scrolled into view on the ios app. 
   The feature to have all of these thumbnails pulled/downloaded to the ios app recursively, or as the title says pre-cached, is a very necessary feature, to avoid waiting for the thumbnails to be downloaded only when the photo is scrolled into view. 
   A simple toggle button in the settings of the ios app would be great to enable or disable this. and if enabled would allow for the user not to have to scroll wait for the thumbnails to show, scroll more then wait for those to populate and so on.

**Comment by Bellusterra** [comment]
#1202 I realized that I had made a similar report 2 years after yours which I will close so there is not a duplicate.

**Comment by cclarke591** [comment]
Using the media tab in the ios app, it seems that once I ran the preview generator commands to create the thumbnails, it did populate the thumbnails (by downloading them from the server). That did take some time though.

I figured, if I set a folder as 'available offline' it would locally generate the thumbnails, but that wasn't the case. At very least have the app generate thumbnails for the 'available offline' folders.

That being said, I feel like this should be a default feature, and not an external app. @ChildLearningClub makes a great point about making a simple toggle button in the settings section, but have that on the server/admin side as well.
